nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,57 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+//Respond with the json for all the meals + meals Parameters
+router.get("/", async (request, response) => {
+  let maxPrice = request.query.maxPrice;
+  let limit = request.query.limit;
+  let createdAfter = request.query.createdAfter;
+  let title = request.query.title;
+  if (maxPrice) {
+    response.send(meals.filter((meal) => meal.price <= maxPrice));
+  } else if (limit) {
+    const limmitedMeals = [];
+    for (let i = 0; i < limit; i++) {
+      limmitedMeals.push(meals[i]);
+    }
+    response.send(limmitedMeals);
+  } else if (createdAfter) {
+    response.send(
+      meals.filter((meal) => meal.createdAt.valueOf() > createdAfter.valueOf())
+    );
+  } else if (title) {
+    response.send(
+      meals.filter((meal) =>
+        meal.title.toLowerCase().includes(title.toLowerCase())
+      )
+    );
+  } else {
+    try {
+      response.send(meals);
+      console.log(meals);
+      console.log("in /api/meals");
+    } catch (error) {
+      throw error;
+    }
+  }
+});
```

Comment:
How your server responds if the request includes more than one valid parameter? for example both maxPrice and createdAfter. Can you change the server such that it takes all the valid parameters into account?